--- cmd/go/build.go
+++ cmd/go/build.go
@@ -1737,6 +1737,7 @@
 	// See http://golang.org/issue/3253.
 	if goos == "darwin" {
 		a = append(a, "-fno-common")
+		a = append(a, "-arch i386 -mmacosx-version-min=10.6")
 	}
 
 	return a
@@ -1768,9 +1769,11 @@
 )
 
 func (b *builder) cgo(p *Package, cgoExe, obj string, gccfiles []string) (outGo, outObj []string, err error) {
+	/*
 	if goos != toolGOOS {
 		return nil, nil, errors.New("cannot use cgo when compiling for a different operating system")
 	}
+	*/
 
 	cgoCFLAGS := stringList(envList("CGO_CFLAGS"), p.CgoCFLAGS)
 	cgoLDFLAGS := stringList(envList("CGO_LDFLAGS"), p.CgoLDFLAGS)
